backend/proxy/proxy.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2019 Shift Devices AG
```

Comment:
`util/` is I think the better location

Also the filename should be the packagename, e.g. `socksproxy/socksproxy.go`